### ISSUE TITLE
[python] Temporarily pin to AnnData 0.9.*

### ISSUE THREAD

**Issue by johnkerl** [single-cell-data/TileDB-SOMA#1765]
As of Friday all Python CI is failing like this:
https://github.com/single-cell-data/TileDB-SOMA/actions/runs/6434852473/job/17474905281

```
E   ImportError: cannot import name 'SparseDataset' from 'anndata._core.sparse_dataset' (/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/anndata/_core/sparse_dataset.py)
```

This is a nuanced issue with a nuanced solution in discussion between @nguyenv , @thetorpedodog , and myself.

It will be dealt with soon, however, to unblock CI and high-pri PRs over the next day or two I will _temporarily_ pin to AnnData 0.9. This issue will be kept open for tracking until the issue is dealt with properly.

**Comment by johnkerl** [comment]
Closing in favor of #1785 for longer-term tracking
